project-api/ballerina-projects/src/main/java/io/ballerina/projects/env/CustomPackageResolver.java

```diff
@@ -65,7 +65,8 @@ public CustomPackageResolver(Project project) {
                 // TODO load the module from the custom_project_dir
                 module = loadFromCustomProjectDir(modLoadRequest);
             }
-            modLoadResponses.add(new ModuleLoadResponse(currentPkg.packageId(), module.moduleId(), modLoadRequest));
+            modLoadResponses.add(new ModuleLoadResponse(module.packageInstance().packageId(), module.moduleId(),
```

Comment:
@sameerajayasoma , using `currentPkg.packageId()` seems to be a bug here which was the issue for package dependencies becoming empty (which we saw during the debug session)

Reply:
I think we should get `packageName` from the `Ballerina.toml` file. At some places, we are taking the project directory name as the default package name. I think this creates errors.

Reply:
AFAIR, here we are only considering the `PackageId`.